### PR TITLE
Fixing #7 and #11

### PR DESCRIPTION
**Pull request by haven1433**
* Changes to a file opened for edit will be reflected in the file without having to reload.
* If there are local changes, they will not be overwritten.
* No undo support: I could do a full file diff here looking for the changes and building up an undo item from it, but there's no good way to do an undo to an edit to the entire file at this point. More interestingly, the undo-history could be used as a way to "merge" the changes: rewind to last saved state, reload the file from disk, and fast-forward changes back to where we were. It's something to consider for a future change if it's requested.
* Proper validation of the model for this simple bugfix required 6 new unit tests: 2 to validate that the model is adding/removing files to be watched properly, 2 to test the reload functionality, and 2 to test the IOException cases.
* 2 additional tests were added to demonstrate the problems with #11, since that was a related issue.

### CODE REVIEW

**Comment by rtzoeller** [comment]
@haven1433 adding a second comment to trigger an email; I modified my previous one in place a few times and I wanted to point out that I found a few crashes not caught in my original review.

**Comment by rtzoeller** [comment]
I don't know why the tracebacks have double `at`s in them, e.g. ` at at HavenSoft.Gen3Hex.View.App.Main()`. I built and ran the program in JetBrains Rider instead of VS; let me know if you need help reproducing any of the crashes.

**Comment by haven1433** [comment]
> I don't know why the tracebacks have double `at`s in them, e.g. ` at at.

I've seen the double ats in other programs before. I think it has something to do with the stack trace getting flattened from exceptions that contain inner exceptions. I'm not going to worry about it.

**Comment by rtzoeller** [comment]
I found a way to reliably reproduce changes not being reflected properly:

1. Create a new file in the editor and modify it to contain some content.
2. Save the file, choosing an arbitrary file name.
3. Create another new file in the editor and modify it with different contents than the first file.
4. Save the new file, choosing to use the same file name as the first file. Confirm replacing it.
5. Switch back to the first file/tab and notice that it has not updated.

**Comment by haven1433** [comment]
> I found a way to reliably reproduce changes not being reflected properly:

Fixing this also fixed #11. Updating the title.

File: Gen3Hex/ViewModel/EditorViewModel.cs [hunk 1]
```diff
@@ -378,6 +381,17 @@ private readonly StubCommand
 
       private void ForwardDelayedWork(object sender, Action e) => RequestDelayedWork?.Invoke(this, e);
 
+      private void TabPropertyChanged(object sender, PropertyChangedEventArgs e) {
+         if (e.PropertyName == nameof(IViewPort.FileName) && sender is IViewPort viewPort) {
+            if (e.PropertyName != nameof(IViewPort.FileName) || !(sender is IViewPort)) return;
```
Comment (rtzoeller) [inline Gen3Hex/ViewModel/EditorViewModel.cs]: The second if statement here seems to contradict the first one.

**Pull request merged**